include/lsst/afw/table/python/sortedCatalog.h

```diff
@@ -1,73 +1,76 @@
 #ifndef AFW_TABLE_PYBIND11_SORTEDCATALOG_H_INCLUDED
 #define AFW_TABLE_PYBIND11_SORTEDCATALOG_H_INCLUDED
-/* 
+/*
  * LSST Data Management System
- * Copyright 2008-2016  AURA/LSST.
- * 
+ * Copyright 2008-2017  AURA/LSST.
+ *
  * This product includes software developed by the
  * LSST Project (http://www.lsst.org/).
  *
  * This program is free software: you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
  * the Free Software Foundation, either version 3 of the License, or
  * (at your option) any later version.
- * 
+ *
  * This program is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU General Public License for more details.
- * 
- * You should have received a copy of the LSST License Statement and 
- * the GNU General Public License along with this program.  If not, 
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
  * see <https://www.lsstcorp.org/LegalNotices/>.
  */
 
-#include <pybind11/pybind11.h>
-//#include <pybind11/operators.h>
-#include <pybind11/stl.h>
+#include "pybind11/pybind11.h"
 
-#include "lsst/afw/table/Source.h"
-#include "lsst/afw/table/Catalog.h"
 #include "lsst/afw/table/SortedCatalog.h"
+#include "lsst/afw/table/python/catalog.h"
 
 namespace lsst {
 namespace afw {
 namespace table {
 namespace python {
 
-/**
-Declare member and static functions for a given instantiation of lsst::afw::table::CatalogT<RecordT>
-that supplement or override methods of lsst::afw::table::CatalogT<RecordT>
+template <typename Record>
+using PySortedCatalog = pybind11::class_<SortedCatalogT<Record>, std::shared_ptr<SortedCatalogT<Record>>,
+                                         CatalogT<Record>>;
 
-To use this:
-- Instantiate a hidden CatalogT<RecordT> unsorted base class, using a name with a leading underscore.
-- Call `declareCatalog` to add methods to the hidden unsorted base class.
-- Instantiate a SortedCatalogT<RecordT> as the class of interest.
-- Call `declareSortedCatalog` to add methods to it.
-- In a python module:
+/**
+Wrap an instantiation of lsst::afw::table::SortedCatalogT<Record>.
 
-    from lsst.afw.table.catalog import addCatalogMethods
-    from lsst.afw.table.sortedCatalog import addSortedCatalogMethods
-    from *yourmodule* import *YourCatalogClass*
-    addCatalogMethods(*YourCatalogClass*)
-    addSortedCatalogMethods(*YourCatalogClass*)
+In addition to calling this method (which also instantiates and wraps the CatalogT base class),
+you must call addCatalogMethods on the class object in Python.
 
-@tparam RecordT  Record type, e.g. BaseRecord or SimpleRecord.
+@tparam Record  Record type, e.g. BaseRecord or SimpleRecord.
```

Comment:
Note this goes against [LSST recommendations](https://developer.lsst.io/coding/cpp_style_guide.html#names-representing-template-parameters-may-be-a-single-uppercase-letter-or-a-mixed-case-phrase-first-letter-capitalized-in-each-word-indicating-the-desired-type) (though I personally prefer the new form).